apps/prairielearn/src/pages/administratorQuery/administratorQuery.html.ts

```diff
@@ -0,0 +1,328 @@
+                      <thead>
+                        <tr>
+                          ${result.columns.map((col) => {
+                            return html` ${renderHeader(result.columns, col)} `;
+                          })}
+                        </tr>
+                      </thead>
+
+                      <tbody>
+                        ${result.rows.map((row) => {
+                          return html`
+                            <tr>
+                              ${result.columns.map((col) => {
+                                return html` ${render(row, col)}`;
+                              })}
+                            </tr>
+                          `;
+                        })}
+                      </tbody>
+                    </table>
+                  </div>
+                `
+              : null}
+          </div>
+
+          <div class="card mb-4">
+            <div class="card-header bg-primary text-white d-flex align-items-center">
+              Recent query runs
+            </div>
+            <div class="table-responsive">
+              ${resLocals.recent_query_runs.length > 0
+                ? html`
+                    <table class="table table-sm">
+                      <thead>
+                        <tr>
+                          <th>Date</th>
+                          <th>Params</th>
+                          <th>User name</th>
+                          <th>User UID</th>
+                        </tr>
+                      </thead>
+                      <tbody>
+                        ${resLocals.recent_query_runs.map((run) => {
+                          return html`
+                            <tr>
+                              <td>
+                                <a href="${`?query_run_id=${run.id}`}"> ${run.formatted_date} </a>
+                              </td>
+                              <td>
+                                <pre class="mb-0">${JSON.stringify(run.params)}</pre>
+                              </td>
+                              <td>${run.user_name}</td>
+                              <td>${run.user_uid}</td>
+                            </tr>
+                          `;
+                        })}
+                      </tbody>
+                    </table>
+                  `
+                : html`
+                    <div class="card-body">
+                      <div class="text-center text-muted">No recent runs found</div>
+                    </div>
+                  `}
+            </div>
+          </div>
+        </main>
+      </body>
+    </html>
+
+    <script>
+      $(function () {
+        $('.tablesorter').tablesorter({
+          theme: 'bootstrap',
+          widthFixed: true,
+          headerTemplate: '{content} {icon}',
+          widgets: ['uitheme', 'zebra'],
+          widgetOptions: {
+            zebra: ['even', 'odd'],
+          },
+        });
+      });
+    </script>
```

Comment:
I know you're just copying existing code here, but generally `<script>` tags shouldn't be outside the root `<html>` element. Consider moving this to be the last child of `<body>` instead.

Reply:
Resolved in commit 12dc009.